language/move-prover/tests/sources/functional/opaque.move

```diff
@@ -0,0 +1,23 @@
+module TestOpaque {
+
+    spec module {
+        pragma verify = true;
+    }
+
+    // Function which has a wrong post condition, so verification fails.
+    fun opaque_incorrect(): u64 {
+        1
+    }
+    spec fun opaque_incorrect {
+        pragma opaque = true;
+        ensures result == 2;
+    }
+
+    fun opaque_caller(): u64 {
+        opaque_incorrect()
+    }
+    spec fun opaque_caller {
+        // because we only use the post condition but not the definition, this should verify
```

Comment:
Perhaps, this should be the opposite? Like:
"we only use the definition but not the post-condition of `opaque_incorrect()`?"

Reply:
Nope it means just this. Here, where we call this function, we only use its post condition. Therefore this verifies OK (even though the definition returns 1).

Reply:
Ah, I now understand it. Thanks.